### PR TITLE
Create 2024-06-03-geoserver-tutorials.md

### PR DESCRIPTION
**Pull request by MiladRafiei**

### CODE REVIEW

File: _posts/2024-06-03-geoserver-tutorials.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+----
+
+### How to Implement Basic Security in Geoserver
+In this session, we want to discuss the Security section in GeoServer, Defining Users, Groups, and Roles, and Granting rights to created users. If you want to access the complete tutorial, simply click on the [link](https://youtu.be/KCTGZJ2Trvw).
+
+[![](https://img.youtube.com/vi/KCTGZJ2Trvw/0.jpg)](https://www.youtube.com/watch?v=KCTGZJ2Trvw)
+
+## Introduction
+GeoServer has a robust security subsystem. Most of the security features are available through the Web Administration interface. In the Security panel, you can find links to set user properties and bind data to security rules. The basic idea is that you create users and roles, and then combine them with data to enable a specific set of access policies. You can also limit read and write access by role.
```
Comment (petersmythe) [inline _posts/2024-06-03-geoserver-tutorials.md:26]: ...combine them with data **rules** to enable...

File: _posts/2024-06-03-geoserver-tutorials.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+To organize your real users, GeoServer provides you with the user, group, and role concepts. With the first two, you can insert real people into the GeoServer security subsystem, and with roles, you can grant rights to real users.
+
+### User definition
+In GeoServer, a user is someone entitled to use the system; it may be another software or a real person. When you add a user to the security system, GeoServer stores a username, uniquely identifying the user, a password, and a set of key/value pairs to store general information about it. You can disable a user at any time, preventing him from using the system.
+
+### Group definition
+A group is a set of users. GeoServer stores a list of usernames belonging to the group and a group name, uniquely identifying the group.
+
+### Roles definition
+GeoServer roles are associated with performing certain tasks or accessing particular resources. Roles are assigned to users and groups, authorizing them to perform the actions associated with the role. So the GeoServer may have many users and it is difficult to assign a role to everyone, therefore, groups can be created and roles can be considered for each group according to the users.
```
Comment (petersmythe) [inline _posts/2024-06-03-geoserver-tutorials.md:40]: > So the GeoServer may have many...

This appears to rather fit into ### Group definition, don't you think?

And also the Note below?

File: _posts/2024-06-03-geoserver-tutorials.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+We just created three users for the three groups and this may seem overkill to you. Consider them as templates for real users. In the real world, we do not want to have too many administrators; we will probably need several "user_readers" and "user_editors" processing the data. Now, we need to define what they can do on GeoServer.
+
+## Defining roles
+A user or a group without any role assigned is useless. Now it is time to create roles and assign them to our users. Please refer to the following points:
+- From the **User, Groups, and Roles** section, select the **Roles** tab. You will find that two roles already exist. They are the administrative roles assigned to the admin account, and they grant access to all GeoServer configurations. Click on the **Edit** link
+- Switch to the **Roles** tab, then click on **Add new role**.
+- Enter `role_reader` as a new role name. We do not need a Parent role. A child role inherits all the grants from the Parent role, making it useful when you want to extend a basic role with more grants. Indeed, we will do this in the next step.
```
Comment (petersmythe) [inline _posts/2024-06-03-geoserver-tutorials.md:61]: Change `grants` -> `rights` (noun)

To grant (verb) a right (noun).

File: _posts/2024-06-03-geoserver-tutorials.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+- The final step is to associate a role to users or groups. Select the **User, Groups, and Roles** page from the left pane, then select the **Groups** list and click on the "group_readers" group to edit it. Add the "role_reader" role to the group and save it.
+- Now click on the "group_editors" group and associate it with the role_editor role.
+- Finally, associate the "group_admins" group to the "role_admin" role.
+
+By defining roles and associating them to the users, we completed the definition of our organization. Now, we need to explore how data is bound to roles and users.
+
+## Accessing data and services
+GeoServer supports access and control, both at the service level, allowing for the lockdown of service operations to only authenticated users who have been granted a particular role, and on a per-layer basis. The two approaches cannot be mixed. If you lock down a service to a role, you cannot grant the access on a specific layer to the same role. When working with layers, you can define rules that specify what a role can do on any specific layer.
```
Comment (petersmythe) [inline _posts/2024-06-03-geoserver-tutorials.md:71]: ` on a per-layer basis`, or per-workspace

`the two approaches cannot be mixed`.  I disagree with this statement.  They are mixed, and they are both considered in series.  Both services rules and data rules must pass.

File: _posts/2024-06-03-geoserver-tutorials.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+The operations controlled are the view, write, and admin access. When granting read access on a layer, you enable a user to add it on a map; while granting write access you enable the user to update, create, and delete features contained in the layer. The admin access level enables the user to update the layer’s configuration.
+
+## Layer Security
+We want to protect the dataset contained in the `test` workspace from unauthorized access while leaving the remaining layers freely available to all users. In this section, we will associate layers and roles:
+- Navigate to the Data > Security page. The rules list shows the two shipped with the default GeoServer configuration.
+The `*.*.r` rule is associated with the `*` roles. 
+This means that "any user", including the anonymous one, can access "any layer" from "any workspace" configured on GeoServer. The general form of the rule is then as in this code: **workspace.layer.accessMode**.
```
Comment (petersmythe) [inline _posts/2024-06-03-geoserver-tutorials.md:79]: `The general form of the rule is then as in this code` -> `The general format of rules is`, or something similar.  `code` is not quite correct here.

**Review by petersmythe — CHANGES_REQUESTED** [review]
Thanks for the post.   I have just made a few suggestions above, including one important correction.

File: _posts/2024-06-03-geoserver-tutorials.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+( [YouTube](https://www.youtube.com/@geospatialtechno)
+| [LinkedIn](https://www.linkedin.com/in/geospatialtechno)
+| [Facebook](https://www.facebook.com/geospatialtechno)
+| [X](https://twitter.com/geospatialtechn)
+)
+
+----
+
+### How to Implement Basic Security in Geoserver
+In this session, we want to discuss the Security section in GeoServer, Defining Users, Groups, and Roles, and Granting rights to created users. If you want to access the complete tutorial, simply click on the [link](https://youtu.be/KCTGZJ2Trvw).
```
[the remark this answers is not in the document]
Comment (MiladRafiei) [reply _posts/2024-06-03-geoserver-tutorials.md:21]: The YouTube video also talks about changing the default password and applying access rights to services, which I have not included in this post due to the length of the content.

File: _posts/2024-06-03-geoserver-tutorials.md (hunk 1 above)
Comment (MiladRafiei) [reply _posts/2024-06-03-geoserver-tutorials.md:26]: It's done.

File: _posts/2024-06-03-geoserver-tutorials.md (hunk 2 above)
Comment (MiladRafiei) [reply _posts/2024-06-03-geoserver-tutorials.md:40]: I fixed it.

File: _posts/2024-06-03-geoserver-tutorials.md (hunk 4 above)
Comment (MiladRafiei) [reply _posts/2024-06-03-geoserver-tutorials.md:71]: I fixed it.

File: _posts/2024-06-03-geoserver-tutorials.md (hunk 5 above)
Comment (MiladRafiei) [reply _posts/2024-06-03-geoserver-tutorials.md:79]: I fixed it.

**Comment by MiladRafiei** [comment]
> Thanks for the post. I have just made a few suggestions above, including one important correction.

Thank you for your feedback. I have implemented your suggestions and corrections.

**Comment by jodygarnett** [comment]
Feedback has been addressed, thanks for adding to content both of you!

**Pull request merged**